api/handle_analytics.go

```diff
@@ -13,14 +13,15 @@ import (
 
 func handleListAnalytics(uc usecases.Usecases) func(c *gin.Context) {
 	return func(c *gin.Context) {
+		ctx := c.Request.Context()
```

Comment:
There is a lot of this change in this PR.
The motivation is the following: if code is executed in the handler _after gin moved on to the next middleware_ (which may happen, for instance because of the timeout middleware) then c.Request.Context() will no longer return the same context (for instance trace and credential keys will be missing).
So the request middleware should be extract as the first thing in the handlers, before taking any possibly long-lived actions.